librz/search/pkey_search.c

```diff
@@ -0,0 +1,151 @@
+// SPDX-FileCopyrightText: 2024 RizinOrg <info@rizin.re>
+// SPDX-FileCopyrightText: 2024 deroad <wargio@libero.it>
+// SPDX-FileCopyrightText: 2008 Nadia Heninger and J. Alex Halderman
+// SPDX-FileCopyrightText: 2008 J. Alex Halderman
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Originally from RSAKeyFinder 1.0 (2008-07-18) By Nadia Heninger and J. Alex Halderman
+// Improved by santitox, jvoisin and spelissier
+
+#include <rz_search.h>
+#include <rz_util.h>
+
+#include "search_internal.h"
+
+/* The minimal length to perform a search is the sizes of
+the sequence tag, the minimal length of the sequence,
+the version marker and the minimal key length. */
+#define PRIVKEY_SEARCH_MIN_LENGTH (1 + 1 + 4 + 1)
+#define KEY_MAX_LEN               26000
+
+typedef struct pkey_marker {
+	const char *metadata;
+	ut8 marker[4];
+} PKeyMarker;
+
+static PKeyMarker rsa_version = {
+	"rsa",
+	/// RFC 3447 for RSA ASN1-bytes representing version field
+	{ 0x02, 0x01, 0x00, 0x02 },
+};
+
+static PKeyMarker ecc_version = {
+	"ecc",
+	/// RFC 5915 for elliptic curves ASN1-bytes representing version field
+	{ 0x02, 0x01, 0x01, 0x04 },
+};
+
+static PKeyMarker safecurves_version = {
+	"safecurves",
+	/// RFC 8410 for Edwards-curve ASN1-bytes representing version field
+	{ 0x02, 0x01, 0x00, 0x30 },
+};
+
+/*Baby BER parser, just good enough for private keys.
```

Comment:
Since there is comment already for this function , maybe convert to Doxygen?

Reply:
this comment is actually wrong. the code is looking for DER encoding, but i will fix this once it will be "ready"